tsdb/engine/tsm1/tombstone.go

```diff
@@ -155,28 +155,30 @@ func (t *Tombstoner) writeTombstone(tombstones []Tombstone) error {
 
 func (t *Tombstoner) readTombstone() ([]Tombstone, error) {
 	f, err := os.Open(t.tombstonePath())
+	if os.IsNotExist(err) {
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
 	defer f.Close()
 
-	if !os.IsNotExist(err) {
-		var b [4]byte
-		_, err := f.Read(b[:])
-		if err != nil {
-			// Might be a zero length file which should not exist, but
-			// an old bug allowed them to occur.  Treat it as an empty
-			// v1 tombstone file so we don't abort loading the TSM file.
-			return t.readTombstoneV1(f)
-		}
+	var b [4]byte
+	_, err = f.Read(b[:])
```

Comment:
@jwilder I didn't notice this before but we should either be checking `n` or using [`io.ReadFull`](https://golang.org/pkg/io/#ReadFull) since we could theoretically get a partial read (or maybe there's a 1, 2, or 3 byte file). Unless that's getting handled in `readTombstoneV1()`.